### DISCUSSION TITLE
[Bug] GitHub API - List workflow runs returns 0 runs even though there are runs

### DISCUSSION THREAD

**Discussion by dominicroessner** [community/community#24626] [Apps, API and Webhooks]
<p>Trying to use the API or the UI to get workflow runs for a specific branch returns no results.</p>
<p>In the UI, when going to Actions on a repo and filtering by branch shows an error message on top of the page <code>We are having problems searching workflow runs. The results may not be complete</code>.</p>
<p>Using the API and the following endpoint:</p>
<pre><code class="lang-bash">curl --location --request GET 'https://api.github.com/repos/{owner}/{repo}/actions/workflows/main.yml/runs?branch=my-branch&amp;per_page=1' \
  -H "Authorization: token &lt;my token&gt;" \
  -H "Content-Type: application/vnd.github.v3+json" \
</code></pre>
<p>Returns the following even though I can clearly see the run through the pull request UI, or when in the Actions UI and not filtering by branch.</p>
<pre><code class="lang-bash">{
  "total_count": 0,
  "workflow_runs": [

  ]
}
</code></pre>

**Comment by ericmatte** [comment]
<p>I have the same problem on my side in a private repository.</p>
<p>If you call <code>https://api.github.com/repos/{owner}/{repo}/actions/workflows/main.yml/runs</code> without the <code>branch</code> parameter, you can clearly see the runs from that branch in the list.</p>
<p>Also, it seems to work when filtering on older branch, but not on the new ones.</p>
<p>I tried using curl and the octokit from <code>@actions/github</code>.</p>

**Comment by alexcampbell86** [comment]
<p>Am seeing a similar issue, i need the <code>status=in_progress</code> parameter for one of my workflows. Without the filter I can see several workflows with a status of <code>in_progress</code>. The filter for <code>status=completed</code> works as expected. But <code>status=in_progress</code> always returns:</p>
<pre><code class="lang-auto">{
  "total_count": 0,
  "workflow_runs": [

  ]
}
</code></pre>

**Comment by torbjornvatn** [comment]
<p>I get this as well, but only for branches created the last couple of days.<br>
If I use the <code>branch=</code> filter for one of my older branches, it works.</p>

**Comment by gat-bryszard** [comment]
<p>Same problem although with <code>status=</code> filter. Described it here - <a href="https://github.community/t/list-workflow-runs-api-not-detecting-in-progress-builds/165029/5" class="inline-onebox">List workflow runs API not detecting "in_progress" builds - #5 by gat-bryszard</a></p>

**Comment by torbjornvatn** [comment, marked as answer]
<p>Seems like it’s resolved now</p>

**Reply by hakim-dc** [reply]
what version you are using pls ?

**Comment by billyvg** [comment]
<p>I have a public repo where this is happening. If I list all workflow runs for a workflow without any query parameters, it returns fine. But if I try to query with a parameter, I always get a 404. Currently using the GH CLI, but also happens in GH Actions.</p>

**Comment by apconole** [comment]
<p>I have a similar problem - for example:</p>
<p>curl -i -s -S -H “Accept: application/vnd.github.v3+json” “<a href="https://api.github.com/repos/ovsrobot/ovs/actions/runs?branch=series_294442" rel="noopener nofollow ugc">https://api.github.com/repos/ovsrobot/ovs/actions/runs?branch=series_294442</a>”</p>
<p>vs.</p>
<p>curl -i -s -S -H “Accept: application/vnd.github.v3+json” “<a href="https://api.github.com/repos/ovsrobot/ovs/actions/runs?branch=series_294634" rel="noopener nofollow ugc">https://api.github.com/repos/ovsrobot/ovs/actions/runs?branch=series_294634</a>”</p>
<p>I am not sure how to proceed</p>

**Comment by BetimShala** [comment]
<p>it’s still happening, how is this resolved?</p>

**Comment by astrelya** [comment]
<p>Hello,</p>
<p>I have the same issue on one of my repo, when I list all runs I can see the one I’m looking for :</p>
<p>“<a href="https://api.github.com/repos/astrelya/REPO/actions/runs%E2%80%9D" rel="noopener nofollow ugc">https://api.github.com/repos/astrelya/REPO/actions/runs”</a><br>
(I find in the run : head_branch=XXX)</p>
<p>but when I specify the branch</p>
<p>“<a href="https://api.github.com/repos/astrelya/REPO/actions/runs?branch=XXX%E2%80%9D" rel="noopener nofollow ugc">https://api.github.com/repos/astrelya/REPO/actions/runs?branch=XXX”</a></p>
<p>I get total_count: 0.</p>
<p>EDIT : when I request with parameter actor=astrelya I don’t get all of the runs neither.<br>
EDIT 2 : I checked on other repos, and It seems that I can’t get any run post April 10th by request with parameters.</p>
<p>Thanks.</p>

**Comment by ivailop** [comment]
<p>In the GitHub UI, simply filtering (e.g. <code>branch: master</code>) Actions returns empty list with a pink-error banner:</p>
<pre><code class="lang-auto">We are having problems searching workflow runs. The results may not be complete.
</code></pre>

**Comment by davinchia** [comment]
<p>We are also seeing this. We are getting inconsistent results from filters. An existing query that has not been modified for months is now returning stale data.</p>

**Comment by kalski** [comment]
<p>We also have the above described problems when using filters.</p>
<p>This bug is marked as solved but it seems to be again open. Is there a way to mark it accordingly here ?</p>

**Comment by flipace** [comment]
<p>Same here, this started to happen somewhen in the past few days, and caused one of our jobs to go rogue and use up all of our “free” GH action minutes during the night until we could stop it :sob:</p>

**Comment by bitops** [comment]
<p>Also seeing this issue when querying the workflow runs with a branch name specified. Thus far it has not been working for 3 days.</p>

**Comment by mimartin12** [comment]
<p>This is still affecting us, not resolved.</p>

**Comment by mimartin12** [comment]
<p>I am also having issues still with this endpoint.</p>
<p>When filtering for the branch with <code>?branch=branch-name</code> I receive</p>
<pre><code class="lang-auto">{
  "total_count": 0,
  "workflow_runs": [

  ]
}
</code></pre>
<p>However, when I just call the <code>/run</code> endpoint, I can see that the branch that I was filtering for is included in that list. Additionally, there is a red banner on our Github projects when filtering for branches in the actions menu.</p>
<blockquote>
<p>We are having problems searching workflow runs. The result may not be complete.</p>
</blockquote>

**Comment by davinchia** [comment]
<p>This is what Github support told me:</p>
<p>On Monday, the search functionality with Actions experienced a breaking error. A fix was implemented soon after the break to address the underlying issue, but now search is being re-indexed which can take up to a few days. While this is happening, un-filtered searches for all workflows and specific workflows are working as expected still, but any filtering on those results is broken at this time.</p>
<p>This is impacting filtering both through the UI and API.</p>
<p>We are closely tracking the re-indexing, and I will follow-up here again once it has completed and search is functioning as expected. In the meantime, please let me know if there is anything further you would like to discuss. Thanks!</p>

**Comment by terrabitz** [comment]
<p>+1 to this issue</p>
<p>We’re still seeing issues when filtering by <code>branch</code>. We see no results returned in our recently-made repository.</p>

**Comment by m0ppers** [comment]
<p>same here:</p>
<pre><code class="lang-auto">curl -u m0ppers:$PAT "https://api.github.com/repos/$OWNER/$REPO/actions/workflows/$ID/runs?page=1&amp;per_page=5&amp;branch=master" | jq '.workflow_runs[] | {created_at: .created_at, head_branch: .head_branch}'
</code></pre>
<p>returns an 8 day old run as first entry</p>
<p>without branch I get the newer results.</p>
<p>I am using <a href="https://github.com/dawidd6/action-download-artifact" class="inline-onebox" rel="noopener nofollow ugc">GitHub - dawidd6/action-download-artifact: A GitHub Action to download an artifact associated with given workflow and commit or other criteria</a> to download a terraform plan artifact from a separate workflow and it currently downloads a stale terraform plan and apllying doesn’t work</p>
<p>:upside_down_face:</p>

**Comment by rajbos** [comment]
This issue is still present, we jus ran into it with filtering on the action runs:

# Get all results: 
`https://api.github.com/repos/owner/repo/actions/runs`
Returns 36 results:
![image](https://user-images.githubusercontent.com/6085745/233736312-53ed7e3f-9206-4adf-b707-507e7a11db6d.png)

# Get only completed:
`https://api.github.com/repos/owner/repo/actions/runs?status=completed`
Returns 22 results: 
![image](https://user-images.githubusercontent.com/6085745/233736440-7568b643-8d9f-4f83-81fc-96d07478e8f1.png)

# Get everything that is `waiting`: 
`https://api.github.com/repos/owner/repo/actions/runs?status=waiting`
![image](https://user-images.githubusercontent.com/6085745/233736539-6fe891a2-6e3f-4908-836a-a0964dd13a45.png)

**Comment by chiragvidani** [comment]
@rajbos We are also facing a similar issue for the last 3 days. `status` filter was working previously, but it is breaking now.
- When adding status filter in API request, its returning 0 runs.
![image](https://user-images.githubusercontent.com/2871708/234204507-1d2be5ff-7679-410c-b128-9a5a2828a6ae.png)


- Without adding status filter in API request, its returning n runs.
![image](https://user-images.githubusercontent.com/2871708/234204543-bfd80b09-684c-4366-9648-cb37efe69092.png)


Can anyone please guide where to report this issue? Also, we are having enterprise license.

**Reply by Lance-Simeon** [reply]
I started seeing it this week as well, in my case it's when filtering by status.

**Comment by viveklak** [comment]
Seeing the same issue as mentioned in https://github.com/orgs/community/discussions/24626#discussioncomment-5690204 - specifically for filters with `status` set to `waiting`. This seems to have started around April 20th.

**Comment by bkgrant** [comment]
Agreed this has recently become an issue for all `queued` or `in_progress` status queries

**Comment by amadosdiop** [comment]
Hello, 
I have the same issue also with filter : `status=pending`

**Comment by metas-ts** [comment]
same here, with `status=completed` i do get my workflow-run (and i see in the JSON-response that it *is* completed ) when i actually drop the `status=completed` from my query

**Reply by metas-ts** [reply]
update: I opened a ticket and got a reply that they identified and addressed a problem with filtered API-results 
So they currently expect filtered&unfiltered API-results to be in sync

**Reply by abhijit-hota** [reply]
Hey @metas-ts Were you using the `created` filter by any chance? Curious as to how you fixed it. I'm facing [a similar issue](https://github.com/orgs/community/discussions/86766).

**Comment by ethitter** [comment]
I'm experiencing a similar issue. I'm trying to collect monthly usage for a specific job, but requests for workflow runs for page 11 or higher return zero results; that's despite the `total_count` showing that there should be results well beyond page 10. I've worked around this by querying for smaller date ranges to ensure that the result set is fewer than 1,000 entries.

**Reply by abhijit-hota** [reply]
Facing a [similar issue](https://github.com/orgs/community/discussions/86766).

> I've worked around this by querying for smaller date ranges to ensure that the result set is fewer than 1,000 entries.

Unfortunately, I need workflow runs from a specific date.

**Comment by abhijit-hota** [comment]
I'm facing an issue with using `created` and `status=completed` filters together. The results are only correct the first 10 pages after which it doesn't return anything. 

I opened a bug report here: #86766

**Comment by KarenJoseph** [comment]
Same issue with check runs api
https://api.github.com/repos/${{ github.repository }}/commits/$COMMIT_SHA/check-runs

where some checks regardless of their state are not listed
while using the ?check_name\=name parameter returns the result

all results are in one page

**Comment by theyoprst** [comment]
Looks like GitHub has broken this completely, it is reproduced in 100% of cases :-(
https://github.com/orgs/community/discussions/194141

**Comment by sxlijin** [comment]
We are seeing this again on 2026-09-04, specifically with the server-side `status=success` filter. The endpoint returns HTTP 200 and valid JSON, but the result set is intermittently stale or incomplete.

Affected query:

```text
GET /repos/BoundaryML/baml/actions/workflows/203063461/runs?branch=canary&event=push&exclude_pull_requests=true&per_page=100&status=success
```

This is also the query generated by:

```bash
gh run list --repo BoundaryML/baml --workflow ci.yaml --branch canary --event push --status success --limit 100
```

We tested the filtered endpoint and the same endpoint without `status=success` back-to-back 10 times, sleeping 60 seconds between each pair. For the unfiltered response we applied `conclusion == "success"` locally.

Results:

- Filtered query was wrong in 3/10 samples.
- Unfiltered query plus local filtering was correct in 10/10 samples.
- The correct filtered response reported `total_count=814` and returned the current successful run first.
- One bad filtered response reported `total_count=742` and returned a page spanning 2026-08-25 back to 2026-07-29.
- Two other bad responses reported `total_count=400` and `530`, returned pages reaching back to May/June, and omitted the newest successful run.
- The unfiltered response consistently reported `total_count=1338`, the same recent page bounds, and the same newest successful candidate in all 10 samples.
- Every bad filtered response was HTTP 200 with valid JSON, so retrying only transport/API errors does not catch this.

**Reply by dslatkin** [reply]
Yeah, we're having the same issue, first noticed on August 27 where we didn't get more than a month of daily workflow runs returned. This was coincidentally timed around a GitHub Actions outage. Then it's happened a few more times the past couple weeks.

We've recently come to the same conclusion independently that in the meantime we need to use `gh run list` without the status filter applied. 🫩

Wouldn't be surprised based on [this comment](https://github.com/orgs/community/discussions/24626#discussioncomment-3244763) if the results just aren't trustworthy/deterministic in general, and that's never been changed or fixed.

**Reply by sxlijin** [reply]
FWIW, I made the original comment using an LLM while I was investigating this (it was the cause of a flaky release pipeline for us), but **we saw the flakiness over all of August**. I didn't bother doing a rigorous historical analysis, so that's an anecdata claim, but we've now been running for the past week without the server-side filtering, having the automation cross-check the commit history to verify query completeness, and haven't seen issues.

If I put on the hat of someone at GH drowning under scaling bugs (and the ridiculous Azure infra migration), then it makes a lot of sense why the server-side filtering isn't high priority for them: the most user-visible features of Actions are (1) the ability to _dispatch_ actions and (2) the UI for browsing them. Consequently, the read query capabilities that are most important for (2) are the ability to fetch mostly consistent, _unfiltered_, most-to-least-recent workflow runs (consistent as in CAP) for _recent_ runs (past 1h, past 6h, past 48h).

Similarly, because the UI for browing Actions results is much more core to GH as a platform offering, then I trust the backend that backs that UI more than I trust `gh run list` filters (I've run into other issues where we'd rather use the API directly instead of the `gh` CLI)

So I'd put money on there being some eventual consistency property of the underlying backend/query system that appears strongly consistent for the unfiltered queries - maybe because of the ordering guarantees or lack of filtering, maybe because of the complexity that would have to get added to the storage/query engine in the cache layers - combined with weak business priorities that mean that `gh run list` with filters just isn't trustworthy.

I'd honestly advocate for them removing those features, or changing them to be implemented as client-side filtering with how broken it is... but that's a different story.
